RATapi/utils/plotting.py

```diff
@@ -222,3 +290,474 @@ def __exit__(self, _exc_type, _exc_val, _traceback):
+        None : do not plot estimated density for this parameter.
+        'normal': normal Gaussian.
+        'lognor': Log-normal probability density.
+        'kernel': kernel density estimation.
+    block : bool, default False
+        Whether Python should block until the plot is closed.
+    return_fig: bool, default False
+        If True, return the figure as an object instead of showing it.
+    hist_settings :
+        Settings passed to `np.histogram`. By default, the settings
+        passed are `bins = 25` and `density = True`.
+
+    """
+
+    # first convert names to indices if given
+    def name_to_index(param: Union[str, int]):
+        """Convert parameter names to indices."""
+        if isinstance(param, str):
+            return results.fitNames.index(param)
+        return param
+
+    if params is None:
+        params = range(0, len(results.fitNames))
+    else:
+        params = list(map(name_to_index, params))
+
+    if estimated_density is not None:
+        temp = {}  # as dictionary cannot change size during iteration
+        for key, value in estimated_density.items():
+            temp[name_to_index(key)] = value
+        estimated_density = temp
+    else:
+        estimated_density = {}
+
+    fig = panel_plot_helper(
+        lambda ax, i: plot_hist(
+            results, i, smooth=smooth, estimated_density=estimated_density.get(i, None), axes=ax, **hist_settings
+        ),
+        params,
+    )
+    if return_fig:
+        return fig
+    fig.show()
+    if block:
+        fig.wait_for_close()
+
+
+@assert_bayesian("Chain")
+def plot_chain_panel(
+    results: RATapi.outputs.BayesResults,
+    params: Union[list[Union[int, str]], None] = None,
+    maxpoints: int = 15000,
+    block: bool = False,
+    return_fig: bool = False,
+):
+    """Plot the MCMC chain for each parameter of a Bayesian analysis.
+
+    Parameters
+    ----------
+    results : RATapi.outputs.BayesResults
+        The results of a Bayesian analysis.
+    params : list[int], default None
+        The indices or names of a subset of parameters if required.
+        If None, uses all indices.
+    maxpoints : int
+        The maximum number of points to plot for each parameter.
+    block : bool, default False
+        Whether Python should block until the plot is closed.
+    return_fig: bool, default False
+        If True, return the figure as an object instead of showing it.
+
+    """
+    chain = results.chain
+    nsimulations, nplots = chain.shape
+    skip = floor(nsimulations / maxpoints)  # to evenly distribute points plotted
+
+    # convert names to indices if given
+    def name_to_index(param: Union[str, int]):
+        """Convert parameter names to indices."""
+        if isinstance(param, str):
+            return results.fitNames.index(param)
+        return param
+
+    if params is None:
+        params = range(0, len(results.fitNames))
+    else:
+        params = list(map(name_to_index, params))
```

Comment:
i remember it breaking something, but it all seems to work now without the `list()`

Reply:
nevermind - it breaks because the panel plot helper needs to get the length of the iterator for the number of plots - `range` has a length attribute but `map` does not!